### PR TITLE
automate release process

### PR DESCRIPTION
**Pull request by kne42**
# Description
Automate the release process using GitHub Actions.

Upon pushing a tag beginning with `v` to `napari/napari`, the CI will:
- copy the release notes to the tag
- build the distribution
- upload the distribution to the appropriate release on GitHub
- upload the distribution to PyPI

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
This is amazing! I love it!! As mentioned in our discussion, one line / a link about how to setup the pypi token on github for authentication added to our release notes would be great.

File: docs/developers/RELEASE.md [hunk 1]
```diff
@@ -49,78 +53,51 @@ Thus, you'll need to tag the [reference](https://git-scm.com/book/en/v2/Git-Inte
...
 
 
-## Tagging the new release candidate
+## tagging the new release candidate
 
 First we will generate a release candidate, which will contain the letters `rc`.
 Using release candidates allows us to test releases on PyPi without using up the actual
 release number.
 
 You should include a basic message with the tag `"Version X.Y.Zrc1"`:
```
Comment (tlambert03) [inline docs/developers/RELEASE.md]: can you comment on whether this is this still necessary, if we're removing the `-m "Version...` bit from the command?

File: .github/workflows/make_release.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        run: |
+          TAG="${GITHUB_REF/refs\/tags\/v/}"  # clean tag
+          VER="${TAG/rc*/}"  # remove pre-release identifier
+          RELEASE_FILEPATH=docs/release/release_${VER//./_}.rst
+          # https://help.github.com/en/actions/reference/workflow-commands-for-github-actions
+          echo "::set-env name=tag::$TAG"
+          echo "::set-env name=release_filepath::$RELEASE_FILEPATH"
+      - name: Create Release
+        id: create_release
+        uses: jbolda/create-release@v1.1.0  # switch to actions/create-release@latest once PR merged
```
Comment (tlambert03) [inline .github/workflows/make_release.yml]: I'm assuming you mean PR https://github.com/actions/create-release/pull/50 ? can you add that url for easy lookup?

**Review by tlambert03 — APPROVED** [review]
looks great!  will be a nice addition to the pipeline :)

File: docs/developers/RELEASE.md (hunk 1 above)
Comment (kne42) [reply docs/developers/RELEASE.md]: good catch! I forgot to remove that, oops!

File: .github/workflows/make_release.yml (hunk 2 above)
Comment (kne42) [reply .github/workflows/make_release.yml]: I included a shorthand instead of the full url, hope that's okay :)

**Comment by kne42** [comment]
@jni When I made this, I didn't realize that we stopped uploading wheels to PyPI. Just because I'm curious, what's the advantage of this? Could you make a suggestion to this PR so that it does not build the wheels? Sorry, I'm not super experienced at packaging! :P

File: .github/workflows/make_release.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        with:
+          python-version: 3.7
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Build Distribution
+        run: |
+          pip install wheel
+          python setup.py sdist bdist_wheel
```
Comment (tlambert03) [inline .github/workflows/make_release.yml:28]:
```suggestion
          python setup.py sdist
```
(if you want to turn off wheels)

**Comment by jni** [comment]
huh? We should definitely upload wheels, though they should be noarch. I can't remember how to limit wheels to certain Python versions, but that should be done also.

**Comment by sofroniewn** [comment]
> huh? We should definitely upload wheels, though they should be noarch

Maybe I've been messing up there, my bad - glad the new process will fix that. I'm very excited about this!!

**Review by jni — APPROVED** [review]
Love this. Thank you @kne42! What do you think about doing the same for scikit-image??? :grimacing::grimacing::grimacing::grimacing:

**Pull request merged**